src/native/corehost/hostmisc/pal.unix.cpp

```diff
@@ -343,80 +343,40 @@ bool is_read_write_able_directory(pal::string_t& dir)
            (access(dir.c_str(), R_OK | W_OK | X_OK) == 0);
 }
 
-bool pal::get_temp_directory(pal::string_t& tmp_dir)
+bool get_extraction_base_parent_directory(pal::string_t& directory)
 {
-    // First, check for the POSIX standard environment variable
-    if (getenv(_X("TMPDIR"), &tmp_dir))
+    // check for the POSIX standard environment variable
+    if (pal::getenv(_X("HOME"), &directory))
     {
-        return is_read_write_able_directory(tmp_dir);
-    }
-
-    // On non-compliant systems (ex: Ubuntu) try /var/tmp or /tmp directories.
-    // /var/tmp is prefered since its contents are expected to survive across
-    // machine reboot.
-    pal::string_t _var_tmp = _X("/var/tmp/");
-    if (is_read_write_able_directory(_var_tmp))
-    {
-        tmp_dir.assign(_var_tmp);
-        return true;
-    }
-
-    pal::string_t _tmp = _X("/tmp/");
-    if (is_read_write_able_directory(_tmp))
-    {
-        tmp_dir.assign(_tmp);
-        return true;
+        return is_read_write_able_directory(directory);
     }
 
     return false;
 }
 
 bool pal::get_default_bundle_extraction_base_dir(pal::string_t& extraction_dir)
 {
-    if (!get_temp_directory(extraction_dir))
+    if (!get_extraction_base_parent_directory(extraction_dir))
     {
         return false;
     }
 
     append_path(&extraction_dir, _X(".net"));
-    pal::string_t dotnetdir(extraction_dir);
-
-    // getuid() is the real user ID, and the call has no defined errors.
-    struct passwd* passwd = getpwuid(getuid());
-    if (passwd == nullptr || passwd->pw_name == nullptr)
-    {
-        return false;
-    }
-
-    append_path(&extraction_dir, passwd->pw_name);
-
     if (is_read_write_able_directory(extraction_dir))
     {
         return true;
     }
 
-    // Create $TMPDIR/.net accessible to everyone
-    if (::mkdir(dotnetdir.c_str(), S_IRWXU | S_IRWXG | S_IRWXO) == 0)
+    // Create $HOME/.net with rwx access to the owner
+    if (::mkdir(extraction_dir.c_str(), S_IRWXU) == 0)
```

Comment:
SDK, SOS etc. uses `$HOME/dotnet` prefixed storage directory, should we use that one instead?  e.g. `$DOTNET_PREFIX/singlefileapps/{app1,app2...}` In certain NFS configurations on linux, `net`, `.net` files are usually placed in HOME dir, so it might cause unexpected conflicts.

Reply:
That's a good suggestion! Would you mind creating an issue for this change? Especially if you have additional details about the `net`/`.net` file usage by other components.

I think we should take this change as is, since it's an important change to move away from temp. Where exactly in home it lands is a somewhat different discussion.

Reply:
I thought this is the first PR adding a new concept of `~/.net`, while a more familiar concept `~/dotnet` exists in the ecosystem. However, it sounds like more work than `sed 's/.net/dotnet/'` on this patch.

Reply:
The same change has been made in 3.1 and 5.0 already (and shipped), so we're just porting it to mainline now.
I'm definitely open to discuss a better location for .NET 6+ if it makes sense.

Reply:
The new location must be under `$HOME`. It can be moved further under `$HOME/dotnet/whatever`, if that is more future-proof. In a separate change.